include/seastar/core/metrics_api.hh

```diff
@@ -355,6 +356,10 @@ public:
     void dirty() {
         _dirty = true;
     }
+
+    std::set<sstring> get_labels() const noexcept {
```

Comment:
This makes a copy of the set to return it. Is that intentional? If not, you can also have this function return a const reference (and be sure that the caller uses the lables immediately).

Reply:
It was intentional, so it would be similar to the way metrics works, but I can probably replace it to const ref becasue I only need the labels when I create the filtering function and not for the entire duration of sending the metrics